expression/builtin_compare.go

```diff
@@ -252,6 +248,13 @@ func (s *builtinCompareSig) eval(row []types.Datum) (d types.Datum, err error) {
 		return
 	}
 
+	if s.op != opcode.NullEQ {
+		if aa, bb, err := types.CoerceDatum(sc, a, b); err == nil {
```

Comment:
now `select * from t where json_field = 0.3` is coverd, but `select * from t where text_field = 0.3` is not. that will be later, after we fix compare text with decimal.